pallets/committee/src/benchmarking.rs

```diff
@@ -0,0 +1,226 @@
+
+use crate::*;
+use frame_benchmarking::benchmarks_instance;
+use frame_support::{dispatch::DispatchResult, traits::UnfilteredDispatchable, StorageValue};
+use polymesh_common_utilities::{
+    benchs::{user, User},
+    MaybeBlock,
+};
+use sp_std::prelude::*;
+
+const PROPOSAL_PADDING_LEN: usize = 10_000;
+const PROPOSALS_NUM: u8 = 100;
+
+fn make_proposal<T, I>(b: u8) -> (<T as Trait<I>>::Proposal, <T as frame_system::Trait>::Hash)
+where
+    I: Instance,
+    T: Trait<I>,
+{
+    let proposal: <T as Trait<I>>::Proposal =
+        frame_system::Call::<T>::remark(vec![b; PROPOSAL_PADDING_LEN]).into();
+    let hash = <T as frame_system::Trait>::Hashing::hash_of(&proposal);
+    (proposal, hash)
+}
+
+fn make_proposals_and_vote<T, I>(users: &[User<T>]) -> DispatchResult
+where
+    I: Instance,
+    T: Trait<I>,
+{
+    ensure!(
+        users.len() > 0,
+        "make_proposals_and_vote requires non-empty users"
+    );
+    for i in 0..PROPOSALS_NUM {
+        let index = Module::<T, I>::proposal_count();
+        let proposal = make_proposal::<T, I>(i + 1).0;
+        identity::CurrentDid::put(users[0].did());
+        Module::<T, I>::vote_or_propose(users[0].origin.clone().into(), true, Box::new(proposal))?;
+        if users.len() > 1 {
+            let hash = *Module::<T, I>::proposals()
+                .last()
+                .ok_or("missing last proposal")?;
+            // cast min(user.len(), N) - 1 additional votes for proposal #N
+            for (j, user) in users.iter().skip(1).take(i as usize).enumerate() {
+                // Vote for the proposal if it's not finalised.
+                if Module::<T, I>::voting(&hash).is_some() {
+                    identity::CurrentDid::put(user.did());
+                    Module::<T, I>::vote(user.origin.clone().into(), hash, index, j % 2 == 0)?;
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+fn make_members_and_proposals<T, I>() -> Result<Vec<User<T>>, DispatchError>
+where
+    I: Instance,
+    T: Trait<I>,
+{
+    let members: Vec<_> = (0..COMMITTEE_MEMBERS_MAX)
+        .map(|i| user::<T>("member", i))
+        .collect();
+    Members::<I>::put(members.iter().map(|m| m.did()).collect::<Vec<_>>());
+    make_proposals_and_vote::<T, I>(&members)?;
+    Ok(members)
+}
+
+fn vote_verify<T, I>(
+    hash: <T as frame_system::Trait>::Hash,
+    ptoposal_num: u32,
+    vote: bool,
+) -> DispatchResult
+where
+    I: Instance,
+    T: Trait<I>,
+{
+    if COMMITTEE_MEMBERS_MAX > 4 || (COMMITTEE_MEMBERS_MAX == 4 && !vote) {
+        // The proposal is not finalised because there is no quorum yet.
+        if let Some(votes) = Voting::<T, I>::get(&hash) {
+            ensure!(votes.index == proposal_num, "wrong proposal_num");
+            if vote {
+                ensure!(votes.ayes.contains(&did), "aye vote missing");
+            } else {
+                ensure!(votes.nays.contains(&did), "nay vote missing");
+            }
```

Comment:
```suggestion
            ensure!(vote == votes.ayes.contains(&did), "aye vote missing");
            ensure!(vote != votes.nays.contains(&did), "nay vote missing");
```

Reply:
There is no need for the second `ensure!`.

Reply:
Well this is a tighter check while also shorter :)

Reply:
Alright :)